### PR TITLE
LUCENE-8927: Set.copyOf and Set.of instead of Collections.unmodifiabl…

### PR DESCRIPTION
**Pull request by atris**

### CODE REVIEW

**Comment by atris** [comment]
cc @jpountz

File: lucene/benchmark/src/java/org/apache/lucene/benchmark/byTask/feeds/DemoHTMLParser.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public InputSource resolveEntity(String publicId, String systemId) {
     }
     
     private static final Set<String> createElementNameSet(String... names) {
-      return Collections.unmodifiableSet(new HashSet<>(Arrays.asList(names)));
+      return Set.copyOf(Arrays.asList(names));
```
Comment (jpountz) [inline lucene/benchmark/src/java/org/apache/lucene/benchmark/byTask/feeds/DemoHTMLParser.java]: `Set.of(names)`?

File: lucene/core/src/java/org/apache/lucene/index/FieldInfos.java [hunk 2]
```diff
@@ -422,6 +422,10 @@ synchronized boolean contains(String fieldName, DocValuesType dvType) {
       }
     }
     
+    synchronized Set<String> getFieldNames() {
+      return Set.copyOf(nameToNumber.keySet());
+    }
```
Comment (jpountz) [inline lucene/core/src/java/org/apache/lucene/index/FieldInfos.java]: we should not resurrect this method that we just removed

File: lucene/benchmark/src/java/org/apache/lucene/benchmark/byTask/feeds/DemoHTMLParser.java (hunk 1 above)
Comment (jpountz) [reply lucene/benchmark/src/java/org/apache/lucene/benchmark/byTask/feeds/DemoHTMLParser.java]: actually we should probably remove this method and fix call sites to use `Set.of` instead?

**Comment by atris** [comment]
@jpountz Thanks, updated the PR

**Comment by jpountz** [comment]
I think you forgot to remove `createElementNameSet`?

**Comment by atris** [comment]
> I think you forgot to remove `createElementNameSet`?

Oops, sorry. Updated

**Comment by munendrasn** [comment]
@atris  @jpountz Is it possible to extend the scope to remove all *Collections.un\** ??
Also, can we extend this change to the Solr project too?

**Comment by atris** [comment]
Solr changes are coming in a separate PR since I did not want to overlap and amplify things in a single PR.

We can explore further changes in subsequent PRs.

**Pull request merged**